### DISCUSSION TITLE
How to get ONLY test failures as output with dotnet test?

### DISCUSSION THREAD

**Discussion by IanKemp** [xunit/xunit#2624] [Question] [answered]
We have a PowerShell script that we use for verifying our code before we push it up to the remote repo and create PRs. It essentially boils down to the following:

```powershell
& dotnet test --nologo --verbosity:quiet $files
```

where `$files` is an array of the unit test DLLs.

The problem with this command is that after it's run all tests, it prints out a summary of the pass/fail counts of every test DLL it ran. We don't want this, it's noise - we just want to be notified if there's a test failure and which test(s) are problematic, or have no output at all if all tests are green. Is there any way to accomplish this?

I've tried using the switches for loggers as documented [here (see `-l|--logger <LOGGER>`)](https://learn.microsoft.com/dotnet/core/tools/dotnet-test) and [here](https://learn.microsoft.com/visualstudio/msbuild/msbuild-command-line-reference#switches-for-loggers) to pass `consoleLoggerParameters`, but trying that:

```powershell
& dotnet test --nologo --verbosity:quiet --logger:"console;consoleLoggerParameters=ErrorsOnly"
```

has no effect.

**Comment by bradwilson** [comment, marked as answer]
We don't control the output of `dotnet test`. This is a question for the VSTest team. https://github.com/microsoft/vstest
